src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsWriteStream.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Data.SqlClientX.IO
+{
+    /// <summary>
+    /// A stream abstraction over the TDS protocol which works to write the TDS packet
+    /// This stream will take care of wrapping the bytes with the TDS header. 
+    /// </summary>
+    internal class TdsWriteStream : Stream, ITdsWriteStream
+    {
+        private Stream _underlyingStream;
+
+        private byte[] _WriteBuffer;
+
+        // Start at the end of the Tds header
+        internal int WriteBufferOffset { get; private set; } = TdsEnums.HEADER_LEN;
+
+        // This should be set before flushing the buffer.
+        public TdsStreamPacketType PacketHeaderType { get; set; }
+
+        internal byte PacketNumber { get; private set; } = 1; // Packets always start with 1.
+
+        public TdsWriteStream(Stream underLyingStream) : base()
```

Comment:
The coding style guide for this project doesn't really specify the above. But it doesn't specify anything :) https://github.com/dotnet/SqlClient/blob/main/coding-style.md 

I can stick to your recommendation, and thank you for it. 

I wish it could be enforceable in VS.

Reply:
I'm not entirely sure, but I've been able to get rider/resharper to yell at me if I put things in the wrong order (well, at least if I put private methods/properties before public ones). But idk if that's a roslyn analyzers thing or what generated it.

Reply:
If this needs to be done right, then we need this enforced in the build pipeline. Doesn't matter if the code author is using Rider/VS/VSCode or Notepad.